### ISSUE TITLE
Reverse proxy broken since last Radarr develop branch update

### ISSUE THREAD

**Issue by masterzxz** [Radarr/Radarr#7005]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When visiting Radarr via the reverse proxy, the connection URL gets rewritten to "localhost:7878".

### Expected Behavior

I expect Radarr to work via the reverse proxy, as it did before.

### Steps To Reproduce

Environment: "develop", version: 4.0.4.5922
In Radarr config: "General"-settings tab, value "Bind Adress" set to "*", value "URL Base" set to "/plx_rdrr"
In NGINX config: `	#SUBDIRECTORY FOR RADARR
		location /plx_rdrr/ {
            root   html;
            proxy_pass http://localhost:7878/plx_rdrr; #INTERNAL IP ADDRESS
			proxy_http_version 1.1;
			proxy_set_header Upgrade $http_upgrade;
			proxy_set_header Connection $http_connection;
        }`
Visit http://redacted.redacted.com/plx_rdrr/, get redirected to http://localhost/plx_rdrr/. The same config works fine for Sonarr, and used to work fine for Radarr till the latest update. Direct access works fine.

### Environment

```markdown
- OS: Windows Server 2019
- Radarr: 4.0.4.5922
- Docker Install: No
- Using Reverse Proxy: Yes
- Browser: Google Chrome 97.0.4692.99 (Official Build) (64-bit)
```

### What branch are you running?

Develop

### Trace Logs?

No relevant information in log files.

[AB#2522](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/2522)

**Comment by RobinDadswell** [comment]
Firstly Radarr != Sonarr, we have updated Radarr to .NET 6 and have a different HTTP Stack, 

Please have a look at the information found here https://wiki.servarr.com/radarr/installation#nginx as it looks like you may have a slightly misconfigured NGINX setup

**Comment by ZuluWhiskey** [comment]
Exactly the same issue here, identical NGINX configuration as was before the upgrade but now it doesn't work

**Comment by ta264** [comment]
In particular Radarr now requires the `Host` and `X-Forwarded-*` headers as seen here: https://github.com/linuxserver/docker-swag/blob/7d6b5e66c1150d53d6b83bfae8f93c1fda8ed90d/root/defaults/proxy.conf#L22-L30

**Comment by ZuluWhiskey** [comment]
Changed my config to follow the linked guide and all working again ☺️

**Comment by masterzxz** [comment]
Changed config as suggested by @RobinDadswell - issue resolved. Thanks for the fast response!

**Comment by Arcanemagus** [comment]
For anyone running HAProxy, the old "cheap" health check of `OPTIONS /` now fails, you need to switch this to `GET /` for HAProxy to consider Radarr to be up.

**Comment by neothematrix** [comment]
hello, sorry to add a comment to a closed issue, but I also had the same issue described above after the last radarr update and I already had the correct nginx configuration, but I kept having issues with the reverse proxy.
My specific issue is that the reverse proxy would redirect me to the wrong port, I use a non-standard https port, i.e.:

`https://my-server.domain.com:9999/radarr`

after radarr upgrade if I tried to connect to the above link, I'd be redirected to: `https://my-server.domain.com/radarr/login?ReturnUrl`

it seemed Radarr didn't properly redirect keeping in mind the non-standard https port.

I even tried adding `proxy_set_header   X-Forwarded-Port 9999;` but it didn's seem to help.

What I ultimately did that fixed my issue was to set:

`proxy_redirect     https://my-server.domain.com/   https://my-server-domain.com:9999/;`

However I'm not sure if this is a fix or a workaround, could this mean radarr is not honoring the original port?

thanks!
